crates/turborepo-ui/src/tui/preferences.rs

```diff
@@ -0,0 +1,132 @@
+use std::{
+    fs::{self, File},
+    io::{BufReader, Write},
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::{from_reader, json, Value};
+use turbopath::AbsoluteSystemPathBuf;
+
+use super::task::TasksByStatus;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Preferences {
+    pub is_task_list_visible: Option<bool>,
+    pub active_task: Option<String>,
+    pub is_pinned_task_selection: Option<bool>,
+}
+
+#[derive(Debug)]
+pub enum PreferenceFields {
+    IsTaskListVisible,
+    ActiveTask,
+    PinnedTaskSelection,
+}
+
+impl Default for Preferences {
+    fn default() -> Self {
+        Self {
+            active_task: None,
+            is_task_list_visible: Some(true),
+            is_pinned_task_selection: Some(false),
+        }
+    }
+}
+
+const TUI_PREFERENCES_PATH_COMPONENTS: &[&str] = &[".turbo", "preferences", "tui.json"];
+
+fn read_json(path: &AbsoluteSystemPathBuf) -> Preferences {
+    File::open(path)
+        .ok()
+        .and_then(|file| from_reader(BufReader::new(file)).ok())
```

Comment:
Not impactful due to how small this JSON file will be, but we should usually always prefer [serde_json::from_str](https://github.com/serde-rs/json/issues/160).